README.md

```diff
@@ -12,62 +12,370 @@ Install from NPM:
+
+```javascript
+class SlowElem extends LitElement {
+  render() {
+    return html`<p>I take my time</p>`;
+  }
+  async getLoadingComplete() {
+    return new Promise(resolve => {
+      setTimeout(() => resolve(), 2000);
+    });
+  }
+}
+```
+
+To bypass waiting for `getLoadingComplete()`, set the `awaitLoadingComplete` configuration option to `false`:
+
+```javascript
+it('should not wait', async() => {
+  const elem = await fixture(html`<slow-elem></slow-elem>`, {
+    awaitLoadingComplete: false
+  });
+});
+```
+
+#### Waiting for a Condition
+
+In cases where there are no other reliable hooks (like events, `getUpdateComplete()` or `getLoadingComplete()`), `waitUntil(condition)` can be used to wait for a particular condition to become `true`. The condition can optionally return a Promise.
+
+```javascript
+import { fixture, waitUntil } from '@brightspace-ui/testing';
+
+it('should wait for condition', async() => {
+  const elem = await fixture(...);
+  await waitUntil(() => elem.foo === 'bar');
+});
+```
+
+By default, `waitUntil` will poll every `50ms` and time out after `1000ms`. Those options can be configured:
+
+```javascript
+await waitUntil(() => elem.condition, {
+  interval: 10,
+  timeout: 2000
+});
+```
+
+> **Note:** because `waitUntil` constantly polls, it can slow down test execution and should be avoided if possible.
+
+### Defining a Custom Element for Testing
+
+If a test requires a one-off custom element, define it using `defineCE` and pass the returned tag name in the call to `fixture()`.
+
+```javascript
+import { defineCE, fixture, html } from '@brightspace-ui/testing';
+
+const tag = defineCE(
+  class extends LitElement {
+    static properties = {
+      foo: { type: String }
+    };
+    render() {
+      return html`hello`;
+    }
+  }
+);
+
+it('should use custom element', async() => {
+  const foo = 'bar';
+  const elem = await fixture(html`<${tag} foo="${foo}"></${tag}>`);
+});
 ```
 
-## Versioning & Releasing
+> **Important:** `defineCE` is not performant and shouldn't be used outside of tests.
+
+### Visual-Diff Testing
+
+Also known as "visual regression" or "perceptual diff", visual-diff testing involves taking snapshot images of the browser and comparing them against a known golden (or "baseline") image. The images are compared pixel-by-pixel and differences beyond a threshold will fail the test. Our visual diff testing leverages the [pixelmatch](https://github.com/mapbox/pixelmatch) library to perfom its comparison.
 
-> TL;DR: Commits prefixed with `fix:` and `feat:` will trigger patch and minor releases when merged to `main`. Read on for more details...
+Unlike other tests, visual-diff tests must be in a file with the `*.vdiff.js` extension.
```

Comment:
```suggestion
Unlike other tests, visual-diff tests must be in a file that matches the pattern `*.vdiff.js`.
```

Reply:
Actually, this isn't quite accurate either since they can defines their own pattern. It just uses `vdiff` as the type _in_ that pattern.

Reply:
Right. So should I just leave this part totally off and it can be discussed down in the "running tests" part?

Reply:
Yeah maybe that's best